compilers/concrete-compiler/compiler/lib/Bindings/Python/CompilerAPIModule.cpp

```diff
@@ -1081,6 +1080,28 @@ void mlir::concretelang::python::populateCompilerAPISubmodule(
       .doc() = "Definition of a partition (in terms of precision in bits and "
                "norm2 in value).";
 
+  // ------------------------------------------------------------------------------//
+  // EXTERNAL PARTITION //
+  // ------------------------------------------------------------------------------//
+  pybind11::class_<concrete_optimizer::utils::ExternalPartition>(
+      m, "ExternalPartition")
+      .def(init([](std::string name, uint64_t macroLog2PolynomialSize,
+                   uint64_t macroGlweDimension, uint64_t macroInternalDim,
+                   double maxVariance, double variance)
+                    -> concrete_optimizer::utils::ExternalPartition {
+             return concrete_optimizer::utils::ExternalPartition{
+                 name,
+                 macroLog2PolynomialSize,
+                 macroGlweDimension,
+                 macroInternalDim,
+                 maxVariance,
+                 variance};
+           }),
+           arg("name"), arg("macro_log2_polynomial_size"),
+           arg("macro_glwe_dimension"), arg("macro_internal_dim"),
+           arg("max_variance"), arg("variance"))
+      .doc() = "Definition of an external partition.";
+
   // ------------------------------------------------------------------------------//
   // KEYSET INFO //
```

Comment:
What about keygen using KeySetInfo instead of ProgramInfo? If it requires a lot of changes in the frontend, then we can think about having both.

Reply:
As mentionned at the end of our discussion, I thought it may incorrectly nudge the user into generating keysets from the virtual keyset info, which would yield a big keygen, while just a small subset would be needed for the circuit.

And thinking about it a bit more, I don't think it would work to use a superset of the necessary keyset on a circuit (nothing fancy, just that we index keys by number, and as such, we may incorrectly lookup keys).

Reply:
Make sense yeah. I'm just thinking how can we showcase the feature then... We just need a way to construct a keyset from a program info using pregenerated keys. If we had a big keyset (with more keys than what the circuit needs), we could have a method that takes the program info and the big keyset, and extract necessary keys with their appropriate index.
We can also simulate all this with a keygen for now, so it's not really necessary, but it will make a better demo with something in this direction.

Reply:
@BourgerieQuentin WDYT ?